spec/unit/provider/zfs/zfs_spec.rb

```diff
@@ -139,7 +139,33 @@
         provider.send("zoned=", "value")
       end
     end
+  end
 
+  context "on Linux" do
+    before do
+      Facter.stubs(:value).with(:operatingsystem).returns("Linux")
+    end
+     it "should get 'acltype' property" do
+       provider.expects(:zfs).with(:get, '-H', '-o', 'value', :acltype, name).returns("value\n")
+       expect(provider.send("acltype")).to eq('value')
+     end
+
+     it "should set acltype=value" do
+       provider.expects(:zfs).with(:set, "acltype=value", name)
+       provider.send("acltype=", "value")
+     end
+  end
 
+  context "when not running Linux" do
+    before do
+      Facter.stubs(:value).with(:operatingsystem).returns("Solaris")
+    end
+    it "should not get 'acltype' property" do
+      provider.expects(:zfs).with(:get, '-H', '-o', 'value', :acltype, name).raises(Puppet::ExecutionFailure, "Failed")
+    end
+
+    it "should not set acltype=value" do
+      provider.expects(:zfs).with(:set, "acltype=value", name).raises(Puppet::ExecutionFailure, "Failed")
```

Comment:
Happy to help! A few things here: you don't need to stub the output of operating system, because that's not going to tell us if `acltype` is a valid property or not. Also, you got the `raises` part of the `expects`, but it's not a puppet failure that will be raised.

This is what I ended up with when I got your tests to work:
```
  describe "acltype" do
    context "when available" do
      it "should get 'acltype' property" do
        provider.expects(:zfs).with(:get, '-H', '-o', 'value', :acltype, name).returns("value\n")
        expect(provider.send("acltype")).to eq('value')
      end
      it "should set acltype=value" do
        provider.expects(:zfs).with(:set, "acltype=value", name)
        provider.send("acltype=", "value")
      end
    end

    context "when not available" do
      it "should get '-' for the acltype property" do
        provider.expects(:zfs).with(:get, '-H', '-o', 'value', :acltype, name).raises(RuntimeError, 'not valid')
        expect(provider.send("acltype")).to eq('-')
      end
      it "should not error out when trying to set acltype" do
        provider.expects(:zfs).with(:set, "acltype=value", name).raises(RuntimeError, 'not valid')
        expect{provider.send("acltype=", "value")}.to_not raise_error
      end
    end
  end
```

We're stubbing the output of zfs to return a runtime error with a message of 'not valid'. We want to make sure the code rescues that error and returns the correct thing. Does that help?